### PR TITLE
Implement BlockJacobi option in ChebyshevSmoother

### PR DESCRIPTION
**Pull request by bergbauer**

### CODE REVIEW

File: include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp [hunk 1]
```diff
@@ -772,8 +772,18 @@ MultigridPreconditionerBase<dim, Number>::initialize_smoother(Operator &   mg_op
...
+        smoothers[level] = std::make_shared<
+          ChebyshevSmoother<Operator, VectorTypeMG, dealii::DiagonalMatrix<VectorTypeMG>>>();
+        initialize_chebyshev_smoother_point_jacobi(mg_operator, level);
+      }
+      else if(data.smoother_data.preconditioner == PreconditionerSmoother::BlockJacobi)
+      {
+        smoothers[level] = std::make_shared<
+          ChebyshevSmoother<Operator, VectorTypeMG, BlockJacobiPreconditioner<Operator>>>();
+        initialize_chebyshev_smoother_block_jacobi(mg_operator, level);
+      }
```
Comment (nfehn) [inline include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp:786]: else-branch with `Assert(false, ExcMessage("Not implemented"));`

File: include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp [hunk 2]
```diff
@@ -846,7 +856,18 @@ MultigridPreconditionerBase<dim, Number>::update_smoother(unsigned int level)
...
+        smoothers[level] = std::make_shared<
+          ChebyshevSmoother<Operator, VectorTypeMG, dealii::DiagonalMatrix<VectorTypeMG>>>();
+        initialize_chebyshev_smoother_point_jacobi(*operators[level], level);
+      }
+      else if(data.smoother_data.preconditioner == PreconditionerSmoother::BlockJacobi)
+      {
+        smoothers[level] = std::make_shared<
+          ChebyshevSmoother<Operator, VectorTypeMG, BlockJacobiPreconditioner<Operator>>>();
+        initialize_chebyshev_smoother_block_jacobi(*operators[level], level);
+      }
```
Comment (nfehn) [inline include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp:870]: as other comment above

File: include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.h [hunk 3]
```diff
@@ -292,7 +292,10 @@ class MultigridPreconditionerBase : public PreconditionerBase<Number>
   initialize_smoother(Operator & matrix, unsigned int level);
 
   void
-  initialize_chebyshev_smoother(Operator & matrix, unsigned int level);
+  initialize_chebyshev_smoother_point_jacobi(Operator & matrix, unsigned int level);
+
+  void
+  initialize_chebyshev_smoother_block_jacobi(Operator & matrix, unsigned int level);
```
Comment (kronbichler) [inline include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.h:298]: This is also present in the `initialize_smoother` function above, but shouldn't we choose `const` arguments in this function?

```suggestion
  initialize_chebyshev_smoother_point_jacobi(Operator & matrix, unsigned int const level);

  void
  initialize_chebyshev_smoother_block_jacobi(Operator & matrix, unsigned int const level);
```

**Review by kronbichler — APPROVED** [review]

File: include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.h (hunk 3 above)
Comment (bergbauer) [reply include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.h]: Yes we should, I updated this PR.

**Pull request merged**